File: 2nd year/MPP/project-mpp/java-project/TriathlonPersistence/src/main/java/triathlon/persistence/validators/AthleteValidator.java
package triathlon.persistence.validators;

import triathlon.model.Athlete;

public class AthleteValidator implements Validator<Athlete> {
    @Override
    public void validate(Athlete athlete) throws ValidationException {
        if (athlete == null) {
            throw new ValidationException("triathlon.model.entities.Entity not instance of athlete");
        }
        if (athlete.getName() == null
                || athlete.getName().matches("^[A-Z]")) {
            throw new ValidationException("Athlete name is invalid (null or malformed)");
        }
    }
}

File: 2nd year/MPP/project-mpp/csharp-project/persistence/validators/AthleteValidator.cs
using System.Text.RegularExpressions;
using model;

namespace persistence.validators
{
    public class AthleteValidator : IValidator<Athlete>
    {
        public void Validate(Athlete athlete)
        {
            if (athlete == null)
            {
                throw new ValidationException("Entity not instance of athlete");
            }

            var nameMatch = Regex.Match(athlete.Name, "^[A-Z]");
            if (athlete.Name == null || !nameMatch.Success)
            {
                throw new ValidationException("Athlete name is invalid (null or malformed)");
            }
        }
    }
}
